env0/resource_environment_test.go

```diff
@@ -486,7 +486,88 @@ func TestUnitEnvironmentResource(t *testing.T) {
 				mock.EXPECT().Template(environment.LatestDeploymentLog.BlueprintId).Times(1).Return(template, nil)
 				mock.EXPECT().EnvironmentCreate(gomock.Any()).Times(1).Return(environment, nil)
 
-				mock.EXPECT().ConfigurationVariablesByScope(client.ScopeEnvironment, environment.Id).Times(1).Return(client.ConfigurationChanges{}, nil) // read after create -> on update -> read after update
+				mock.EXPECT().ConfigurationVariablesByScope(client.ScopeEnvironment, environment.Id).Times(1).Return(client.ConfigurationChanges{configurationVariables}, nil) // read after create -> on update -> read after update
+				mock.EXPECT().Environment(environment.Id).Times(1).Return(environment, nil)
+
+				mock.EXPECT().EnvironmentDestroy(environment.Id).Times(1)
+			})
+		})
+
+		// Tests use-cases where the response returned by the backend varies from the order of the state.
+		t.Run("Create unordered configuration variables", func(t *testing.T) {
```

Comment:
The state is order is conf1 conf2.
The API response is conf2 conf1.

This test failed without my fix. After my fix this test passes.